hwconf/Zerno/hw_zerno_drive_core.c

```diff
@@ -36,727 +36,810 @@
+                   PAL_STM32_OSPEED_HIGHEST |
+                   PAL_STM32_PUDR_FLOATING );
+    palSetPadMode( GPIOA, 9, PAL_MODE_ALTERNATE( GPIO_AF_TIM1 ) |
+                   PAL_STM32_OSPEED_HIGHEST |
+                   PAL_STM32_PUDR_FLOATING );
+    palSetPadMode( GPIOA, 10, PAL_MODE_ALTERNATE( GPIO_AF_TIM1 ) |
+                   PAL_STM32_OSPEED_HIGHEST |
+                   PAL_STM32_PUDR_FLOATING );
+
+    palSetPadMode( GPIOB, 13, PAL_MODE_ALTERNATE( GPIO_AF_TIM1 ) |
+                   PAL_STM32_OSPEED_HIGHEST |
+                   PAL_STM32_PUDR_FLOATING );
+    palSetPadMode( GPIOB, 14, PAL_MODE_ALTERNATE( GPIO_AF_TIM1 ) |
+                   PAL_STM32_OSPEED_HIGHEST |
+                   PAL_STM32_PUDR_FLOATING );
+    palSetPadMode( GPIOB, 15, PAL_MODE_ALTERNATE( GPIO_AF_TIM1 ) |
+                   PAL_STM32_OSPEED_HIGHEST |
+                   PAL_STM32_PUDR_FLOATING );
+
+
+    // Hall sensors
+    palSetPadMode( HW_HALL_ENC_GPIO1, HW_HALL_ENC_PIN1, PAL_MODE_INPUT );
+    palSetPadMode( HW_HALL_ENC_GPIO2, HW_HALL_ENC_PIN2, PAL_MODE_INPUT );
+    palSetPadMode( HW_HALL_ENC_GPIO3, HW_HALL_ENC_PIN3, PAL_MODE_INPUT );
+
+    // Phase filters
+
+    palSetPadMode( PHASE_FILTER_GPIO, PHASE_FILTER_PIN,
+                   PAL_MODE_OUTPUT_PUSHPULL |
+                   PAL_STM32_OSPEED_HIGHEST );
+    PHASE_FILTER_OFF();
+
+    // Sensor port voltage
+    SENSOR_PORT_3V3();
+    palSetPadMode( SENSOR_VOLTAGE_GPIO, SENSOR_VOLTAGE_PIN,
+                   PAL_MODE_OUTPUT_PUSHPULL | PAL_STM32_OSPEED_HIGHEST );
+
+    // ADC Pins
+    palSetPadMode( GPIOA, 0, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOA, 1, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOA, 2, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOA, 3, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOA, 5, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOA, 6, PAL_MODE_INPUT_ANALOG );
+
+    // Switch input pins
+    palSetPadMode( HW_SW_PORT, HW_SW_PIN, PAL_MODE_INPUT_PULLUP );
+    palSetPadMode( HW_MOMENTARY_PORT, HW_MOMENTARY_PIN, PAL_MODE_INPUT_PULLUP );
+
+    // PFC interface signals
+    palSetPadMode( PFC_STATUS_PORT, PFC_STATUS_PIN, PAL_MODE_INPUT );
+    palSetPadMode( PFC_ENABLE_PORT, PFC_ENABLE_PIN, PAL_MODE_OUTPUT_PUSHPULL );
+
+    palSetPadMode( GPIOB, 1, PAL_MODE_INPUT_ANALOG );
+
+    palSetPadMode( GPIOC, 0, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOC, 1, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOC, 2, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOC, 3, PAL_MODE_INPUT_ANALOG );
+    palSetPadMode( GPIOC, 4, PAL_MODE_INPUT_ANALOG );
+
+    palSetPadMode( GPIOA, 4, PAL_MODE_INPUT_ANALOG );
+
+    mc_interface_set_pwm_callback( adc_read_callback );
+
+    define_default_values();
+
+    encoder_calibrate_offset();
+
+    if( !speed_thread_running )
+    {
+        chThdCreateStatic( speed_thread_wa, sizeof( speed_thread_wa ), NORMALPRIO, speed_thread, NULL );
+        speed_thread_running = true;
+    }
```

Comment:
```suggestion
chThdCreateStatic( speed_thread_wa, sizeof( speed_thread_wa ), NORMALPRIO, speed_thread, NULL );
```

Why it should have a flag in order to call this ? If `hw_init_gpio` is only called once, there no need to use `speed_thread_running`.

Reply:
Sure, just wanted to be sure that the threads start running after the gpio init.